### PR TITLE
#1307 : Changed dateReviewed to dateImplemented

### LINKED ISSUE #1307: Change requests expire 5 days after implemented instead of after approved
### Description

Currently, change change requests become unusable 5 days after being approved. We wanna change this to be 5 days after being implemented, so that it still restricts change requests from being incorrectly reused for changes they shouldn't be used for, but allows a user more leeway for using an approved change request.

### Acceptance Criteria

The backend checks for when a change request should be invalid should be updated to make a change request invalid when it has an associated change that was implemented more than 5 days ago.
The frontend places where we have an autocomplete with change request options should also be updated to display the correct choices.

### Proposed Solution

Change the code from checking the date reviewed to checking the oldest date implemented on the associated changes for a change request.

### Mocks

N/A

### PR DESCRIPTION
**Pull request by martin0he**
## Changes

In both the back-end and front-end, I changed dateReviewed to dateImplemented by accessing the changeRequest's first change and using its Date.

## Notes

I wasn't able to test it properly using prisma studio so i wasn't able to make sure the front-end autocomplete works as it should

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #1307

### CODE REVIEW

**Review by walker-sean — APPROVED** [review]
LGTM

File: src/backend/src/utils/change-requests.utils.ts [hunk 1]
```diff
@@ -134,15 +134,15 @@ export const updateBlocking = async (
...
+  const changeRequest = await prisma.change_Request.findUnique({ where: { crId }, include: { changes: true } });
   const currentDate = new Date();
 
   if (!changeRequest) throw new NotFoundException('Change Request', crId);
   if (changeRequest.dateDeleted) throw new HttpException(400, 'Cannot use a deleted change request!');
   if (changeRequest.accepted === null) throw new HttpException(400, 'Cannot implement an unreviewed change request');
   if (!changeRequest.accepted) throw new HttpException(400, 'Cannot implement a denied change request');
   if (!changeRequest.dateReviewed) throw new HttpException(400, 'Cannot use an unreviewed change request');
-  if (currentDate.getTime() - changeRequest.dateReviewed.getTime() > 1000 * 60 * 60 * 24 * 5)
+  if (currentDate.getTime() - changeRequest.changes[0].dateImplemented.getTime() > 1000 * 60 * 60 * 24 * 5)
```
Comment (RChandler234) [inline src/backend/src/utils/change-requests.utils.ts:145]: you should abstract the method used in the change request transformer to calculate the date implemented and use it here so we're calculating it the same way in both places.

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by martin0he** [comment]
The function to get the dateImplemented has now been abstracted across two files.

**Review by RChandler234 — CHANGES_REQUESTED** [review]
Looking good!

File: src/backend/src/utils/change-requests.utils.ts [hunk 2]
```diff
@@ -134,15 +134,16 @@ export const updateBlocking = async (
...
   const currentDate = new Date();
 
   if (!changeRequest) throw new NotFoundException('Change Request', crId);
   if (changeRequest.dateDeleted) throw new HttpException(400, 'Cannot use a deleted change request!');
   if (changeRequest.accepted === null) throw new HttpException(400, 'Cannot implement an unreviewed change request');
   if (!changeRequest.accepted) throw new HttpException(400, 'Cannot implement a denied change request');
   if (!changeRequest.dateReviewed) throw new HttpException(400, 'Cannot use an unreviewed change request');
-  if (currentDate.getTime() - changeRequest.dateReviewed.getTime() > 1000 * 60 * 60 * 24 * 5)
+  const dateImplemented = getDateImplemented(changeRequest);
+  if (dateImplemented && currentDate.getTime() - dateImplemented.getTime() > 1000 * 60 * 60 * 24 * 5)
```
Comment (RChandler234) [inline src/backend/src/utils/change-requests.utils.ts:146]: this should actually be <, to get the oldest dateImplemented. I'm not sure why we were getting the most recent date implemented before in the transformer

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/backend/src/utils/change-requests.utils.ts (hunk 2 above)
Comment (martin0he) [reply src/backend/src/utils/change-requests.utils.ts:146]: Ok, transformer has now been fixed to get the oldest date.

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**